### PR TITLE
REST fetch

### PR DESCRIPTION
**Pull request by bumblebot1**
Replaced the XMLHttpRequests with fetch calls

### CODE REVIEW

File: public/main.js [hunk 1]
```diff
@@ -48,24 +48,101 @@ form.onsubmit = function(event) {
...
 };
 
-function createTodo(title, callback) {
-    var createRequest = new XMLHttpRequest();
-    createRequest.open("POST", "/api/todo");
-    createRequest.setRequestHeader("Content-type", "application/json");
-    createRequest.send(JSON.stringify({
-        title: title,
+//the flag parameter is set when the response body should be use for the callback
+function responseHandlerFactory(callback, err, status, flag) {
```
Comment (lclarksl) [inline public/main.js]: `flag` could be better named

File: public/main.js [hunk 2]
```diff
@@ -48,24 +48,101 @@ form.onsubmit = function(event) {
...
+    callback = callback || reloadTodoList;
+
+    fetch("/api/todo/" + todo.id, {
+        method: "DELETE",
+        headers: {
+            "Content-type": "application/json"
+        }
+    })
+    .then(
+        responseHandlerFactory(callback, "Failed to delete. Server returned ", 200, false)
```
Comment (cuthullu) [inline public/main.js]: There's a `catch` method on the fetch api so you can split these here something like

```
fetch()
    .then(handleData)
    .catch(handleError)
```

Comment (cuthullu) [inline public/main.js]: Eh never mind. There is a `catch` but it isn't thrown automatically for bad responses

File: public/main.js [hunk 3]
```diff
@@ -48,24 +48,114 @@ form.onsubmit = function(event) {
...
+        method: "POST",
+        headers: {
+            "Content-type": "application/json"
+        },
+        body: JSON.stringify(payload)
+    })
+    .then(
+        responseStatusCheck(201, "Failed to create item. Server returned ")
+    )
+    .then(function(res) {
```
Comment (cuthullu) [inline public/main.js]: This can be squashed to a single line 'then(callback)'

Comment (bumblebot1) [inline public/main.js]: true ill change that now

**Comment by cuthullu** [comment]
:shipit:

**Pull request merged**